wagtail/contrib/settings/views.py

```diff
@@ -172,3 +175,42 @@ def get_success_message(self):
             _("%(setting_type)s updated.")
             % {"setting_type": self.model._meta.verbose_name}
         )
+
+    def get_preview_url(self):
+        if isinstance(self.object, PreviewableMixin) and self.object.is_previewable():
+            return reverse(
+                self.preview_url_name,
+                args=(self.app_name, self.model_name, self.pk),
+            )
+
+    def get_side_panels(self):
+        side_panels = super().get_side_panels()
+        if preview_url := self.get_preview_url():
+            side_panels.append(
+                PreviewSidePanel(self.object, self.request, preview_url=preview_url)
+            )
+            side_panels.append(ChecksSidePanel(self.object, self.request))
+        return side_panels
+
+
+class PreviewOnEdit(preview.PreviewOnEdit):
+    def setup(self, request, app_name, model_name, *args, **kwargs):
+        self.app_name = app_name
+        self.model_name = model_name
+        self.model = get_model_from_url_params(app_name, model_name)
+        self.pk = kwargs.get("pk")
+        super().setup(request, app_name, model_name, *args, **kwargs)
+
+    def get_object(self, queryset=None):
+        self.site = None
+        if issubclass(self.model, BaseSiteSetting):
+            self.site = get_object_or_404(Site, pk=self.pk)
+            return self.model.for_site(self.site)
+        else:
+            return get_object_or_404(self.model, pk=self.pk)
+
+    def get_extra_request_attrs(self):
+        attrs = super().get_extra_request_attrs()
+        # Inject current instance into request so the context processor uses it
+        attrs[self.object.get_cache_attr_name()] = self.object
+        return attrs
```

Comment:
With some small refactoring in the generic `PreviewOnEdit` view, I was able to inject the instance to the request (see 79891f5cdc6736e70cc5fdd962da063bd7847e84). This means we can use `PreviewableMixin` as-is instead of having to use a separate `PreviewableSettingMixin` @SebCorbin 🥳